.cirrus.yml

```diff
@@ -262,3 +243,53 @@ task:
   test_script:
     - ./ci/cirrus.sh
   << : *CAT_LOGS
+
+# Sanitizers
+task:
+  container:
+    dockerfile: ci/linux-debian.Dockerfile
+    cpu: 1
+    memory: 1G
+  env:
+    BUILD:
+    ECDH: yes
+    RECOVERY: yes
+    EXPERIMENTAL: yes
+    SCHNORRSIG: yes
+    CTIMETEST: no
+    EXTRAFLAGS: "--disable-openssl-tests"
+  matrix:
+    - name: "Valgrind (memcheck)"
+      env:
+        # The `--error-exitcode` is required to make the test fail if valgrind found errors, otherwise it'll return 0 (https://www.valgrind.org/docs/manual/manual-core.html)
+        WRAPPER_CMD: "valgrind --error-exitcode=42"
+    - name: "UBSan, ASan, LSan"
+      env:
+        # Use dummy wrapper command to run the tests outside of make check
```

Comment:
Remind me, what's the reason for not running this (and valgrind) in `make check`? If it was just about being able to set the test "count"/TEST_ITERS separately then we do have an env var for this now, and could remove it like this https://github.com/jonasnick/secp256k1/commits/202010_ci_matrix-jn (untested).

Reply:
I think it's just that `make check` will try to run without the wrappers.

Reply:
You mean `make check` is run without a wrapper even when a wrapper is set? Afaics no, because we unset `BUILD` whenever we set `WRAPPER_CMD`.

Reply:
The main issue (and I think this is what Pieter is partly saying) is that `${WRAPPER_CMD} make check` typically won't work for all wrapper commands. 
 * `valgrind make check` will memcheck `make` but will not `memcheck` our tests because they'll be child processes. (Now you could try `valgrind --trace-children` but then it will still additionally memcheck `make` which gives false positives and wastes CI time.)  
 * `wine make check` just won't work at all because `make` is not a Windows executable.
 * I haven't tried QEMU but I expect the same as for Wine (because we cross-build).

What we could is to tell `make` to use the wrapper command. Now that I'm reading up on this again, I recall that it's not even more work to do this. There's the (obscurely named) `LOG_COMPILER` env variable, which does exactly this. You can even pass `LOG_COMPILER="./libtool --mode=execute valgrind"` etc, which we don't need now but maybe in the future. Running all tests using `make check` will have the benefit that the logs always end up in the same place in the CI output. Let me give it a shot.  

And thanks for reminding me of `SECP256K1_TEST_ITERS`. When I was writing this, I didn't know that this exists (I reinvented the name `TEST_ITERS`). So we should simply set this variable instead, like we do for the benchmarks. This has the nice side effect that it would apply to the ordinary `make check` builds if we'd like to set it there for some reason.